### PR TITLE
Allow passing multiple gaurds to the auth middleware

### PR DESCRIPTION
**Pull request by JosephSilber**
Using this, you'd be able to allow access to a single route for multiple guards:

```php
Route::get('foo', 'Foo@foo')->middleware('auth:web,api');
```

### CODE REVIEW

**Comment by taylorotwell** [comment]
I think the check might should be changed to `if (! $this->authenticated())` because the `guest` method reads weird how it returns false, etc.

**Comment by taylorotwell** [comment]
Thoughts?

**Comment by JosephSilber** [comment]
@taylorotwell updated.

File: app/Http/Middleware/Authenticate.php [hunk 1]
```diff
@@ -12,19 +12,40 @@ class Authenticate
...
      */
-    public function handle($request, Closure $next, $guard = null)
+    public function handle($request, Closure $next, ...$guards)
     {
-        if (Auth::guard($guard)->guest()) {
-            if ($request->ajax() || $request->wantsJson()) {
-                return response('Unauthorized.', 401);
-            } else {
-                return redirect()->guest('login');
+        if ($this->check($guards)) {
```
Comment (gaomd) [inline app/Http/Middleware/Authenticate.php]: Looks weird to put success condition before failed condition, in Middleware.

Comment (JosephSilber) [inline app/Http/Middleware/Authenticate.php]: It's not weird at all. You just maybe haven't seen it be done that way.

I actually like it quite a bit. It decreases the level of indentation, which is always a good thing.

**Comment by taylorotwell** [comment]
Would anyone ever want this to be an "and" check instead of an "or" check?

**Comment by JosephSilber** [comment]
For an "and" check, you can just run the middleware twice:

```php
->middleware('auth|auth:api');
```

**Comment by shehi** [comment]
Make it OR please. Not every JSON request is an AJAX request.

**Comment by shehi** [comment]
And why "allow" has become synonymous with "make it mandatory"?! Why ```$guards``` parameter is mandatory?

**Comment by taylorotwell** [comment]
`$guards` isn't mandatory.

**Pull request merged**